lexicon/modification.py

```diff
@@ -210,6 +210,32 @@ def __init__(self, bliss_lexicon: tk.Path, nonword_phones: Optional[List] = None
     def tasks(self):
         yield Task("run", mini_task=True)
 
+    def _eow_phoneme(self, phoneme: str) -> str:
+        """
+        Creates the EOW version of a given phoneme.
+        """
+        return phoneme + self.boundary_marker
+
+    def _modify_pronunciation(self, pronunciation: str, special_phonemes: Set[str]) -> str:
+        """
+        Find the rightmost phoneme in the pronunciation that is not special and replace it by its
+        EOW version. Might do nothing if all phonemes are special.
+
+        Example: "AA BB [noise]" -> "AA BB# [noise]"
+
+        :param pron: Original pronunciation as a string containing phonemes separated by whitespaces
+        :param special_phonemes: Set of special phonemes that should be skipped over.
+        :return: Modified pronunciation
+        """
+        phoneme_list = pronunciation.split()
+
+        for i, phoneme in reversed(list(enumerate(phoneme_list))):
+            if phoneme in special_phonemes:
+                continue
+            phoneme_list[i] = self._eow_phoneme(phoneme)
```

Comment:
```suggestion
            phoneme_list[i] = self._eow_phoneme(phoneme)
            break
```
I think you are currently adding eow marker to all non-special phonemes. 
Fix this by ending the for-loop after first match.